docs/using-react-redux/connect-dispatching-actions-with-mapDispatchToProps.md

```diff
@@ -0,0 +1,419 @@
+This is the only way to trigger a state change.
+
+With React-Redux, your components never access the store directly - `connect` does it for you.
+React-Redux gives you two ways to let components dispatch actions:
+
+- By default, a connected component receives `props.dispatch` and can dispatch actions itself.
+- `connect` can accept an argument called `mapDispatchToProps`, which lets you create functions that dispatch when called, and pass those functions as props to your component.
+
+The `mapDispatchToProps` functions are normally referred to as `mapDispatch` for short, and of course the actual variable name doesn’t matter in real code.
+
+## With or Without `mapDispatchToProps`
+
+### Default: `dispatch` as a Prop
+
+If you don't specify the second argument to `connect()`, your component will receive `dispatch` by default. For example:
+
+```js
+connect()(MyComponent);
+// which is equivalent with
+connect(
+  null,
+  null
+)(MyComponent);
+
+// or
+connect(mapStateToProps /** no second argument */)(MyComponent);
+```
+
+Once you have connected your component in this way, your component receives `props.dispatch`. You may use it to dispatch actions to the store.
+
+```js
+function Counter({ count, dispatch }) {
+  return (
+    <div>
+      <button onClick={() => dispatch({ type: "DECREMENT" })}>-</button>
+      <span>{count}</span>
+      <button onClick={() => dispatch({ type: "INCREMENT" })}>+</button>
+      <button onClick={() => dispatch({ type: "RESET" })}>reset</button>
+    </div>
+  );
+}
+```
+
+### Providing A `mapDispatchToProps`
+
+Providing a `mapDispatchToProps` allows you to specify which actions your component might need to dispatch. It lets you provide action dispatching functions as props. Therefore, instead of calling `props.dispatch(() => increment())`, you may call `props.increment()` directly. There are a few reasons why you might want to do that.
+
+#### More Declarative
+
+First, encapsulating the dispatch logic into function makes the implementation more declarative.
+Dispatching an action and let the Redux store handle the data flow is _how to_ implement the feature, it is not _what_ it does.
+Therefore, instead of saying:
+
+```js
+<button onClick={() => dispatch({ type: "SOMETHING" })} />
+```
+
+you may prefer saying:
+
+```js
+<button onClick={doSomething} />
+```
+
+Once you've wrapped all our action creators with functions that dispatch the actions, the component is free of the need of `dispatch`.
+Therefore, **if you define your own `mapDispatchToProps`, the connected component will no longer receive `dispatch`.**
+
+#### Pass Down Action Dispatching Logic to ( Unconnected ) Child Components
+
+Furthermore, you also gain the freedom of passing down the action dispatching functions to child ( likely unconnected ) components. Now, both the connected component and its descendant need not be aware of `dispatch`, while performing their communications with the Redux store.
```

Comment:
```suggestion
In addition, you also gain the ability to pass down the action dispatching functions to child ( likely unconnected ) components. This allows more components to dispatch actions, while keeping them "unaware" of Redux.  A good example would be dispatching an action when a button is clicked.  Connecting the button directly probably doesn't make sense conceptually, and neither does having the button reference `dispatch`.  Instead, we just want to use `<button onClick={increment} />`.
```

Reply:
Hah, I wrote this suggestion and then went back and saw you'd said basically the same thing in the previous heading :)

Tell you what.  Let's keep the "Declarative" section as just the couple paragraphs, and move all the `<button>` examples into this heading.

Reply:
What about like this :)